hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/dn/ratis/TestDnRatisRandomRead.java

```diff
@@ -0,0 +1,203 @@
+
+    // Set the fs.defaultFS and start the filesystem
+    conf.set(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY, rootPath);
+    // Set the number of keys to be processed during batch operate.
+    conf.setInt(OZONE_FS_ITERATE_BATCH_SIZE, 5);
+
+    fs = FileSystem.get(conf);
+
+  }
+
+  @AfterClass
+  public static void teardown() {
+    IOUtils.closeQuietly(client);
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+    IOUtils.closeQuietly(fs);
+  }
+
+  @Before
+  public void createVolumeAndBucket() throws IOException {
+    // create a volume and a bucket to be used by RootedOzoneFileSystem (OFS)
+    OzoneBucket bucket =
+        TestDataUtil.createVolumeAndBucket(client, BUCKET_LAYOUT);
+    volumeName = bucket.getVolumeName();
+    volumePath = new Path(OZONE_URI_DELIMITER, volumeName);
+    bucketName = bucket.getName();
+    bucketPath = new Path(volumePath, bucketName);
+  }
+
+  @After
+  public void cleanup() throws IOException {
+    fs.delete(bucketPath, true);
+    fs.delete(volumePath, false);
+  }
+
+  @Test
+  public void testRandomRead()
+      throws Exception {
+    String key = "ratiskeytest";
+    int numFiles = 100;
+    // write files
+    for (int i = 0; i < numFiles; i++) {
+      try (OzoneOutputStream outputStream = client.getObjectStore()
+          .getVolume(volumeName)
+          .getBucket(bucketName)
+          .createKey(key + i, i * 100,
+              RatisReplicationConfig.getInstance(
+                  HddsProtos.ReplicationFactor.THREE),
+              new HashMap<>())) {
+        outputStream.write(RandomUtils.nextBytes(i));
+      }
+    }
+    // read files
+    for (int i = 0; i < numFiles; i++) {
+      try (
+          OzoneInputStream ozoneInputStream = client.getObjectStore()
+              .getVolume(volumeName)
+              .getBucket(bucketName)
+              .readFile(key + i)) {
+        ozoneInputStream.read();
+      }
+    }
+    // collect metrics from datanode
+    List<Long> transactionsCountByDn = new ArrayList<>();
+    cluster.getHddsDatanodes().forEach(dn -> {
+      long metrics =
+          dn.getDatanodeStateMachine().getContainer().getMetrics()
+              .getContainerOpsMetrics(
+                  ContainerProtos.Type.GetBlock);
+      if (metrics > 0) {
+        transactionsCountByDn.add(metrics);
+      }
+      System.out.println(dn.getDatanodeDetails().getUuid() + " " + metrics);
+    });
+    // read transactions must exist in all three DN
+    assertEquals(REPLICATION_COUNT, transactionsCountByDn.size());
+    long avgTransactionCountPerNode = numFiles / REPLICATION_COUNT;
+    // expected deviation between transactions count by each datanode is
+    // 1/3 number of files per each datanode (3)
+    long deviation = numFiles / REPLICATION_COUNT / 3;
+    transactionsCountByDn.forEach(
+        count -> assertTrue(
+            Math.abs(count - avgTransactionCountPerNode) < deviation));
```

Comment:
This test is flaky: failed in [5% of repeated runs](https://github.com/adoroszlai/ozone/actions/runs/6835828604/job/18590315506#step:3:19).

Reply:
That's true, I tried to check some uniform distribution for transactions, but seems that it is not possible from this test. Will remove deviation check.

Reply:
Thanks, it's improved, no failures in [100 runs](https://github.com/adoroszlai/ozone/actions/runs/6859269045).